lib/puppet/provider/package/gem.rb

```diff
@@ -124,13 +130,20 @@ def install(useversion = true)
           command << source
         else
           # interpret it as a gem repository
-          command << "--source" << "#{source}" << resource[:name]
+          command << "--source" << "#{source}"
         end
       end
+    end
+
+    command_version = gem_version(command(:gemcmd))
+    if Puppet::Util::Package.versioncmp(command_version, '2.0.0') == -1
+      command << '--no-rdoc' << '--no-ri'
     else
-      command << "--no-rdoc" << "--no-ri" << resource[:name]
+      command << '--no-document'
```

Comment:
Rubygems added [--no-document in 2.0](https://github.com/rubygems/rubygems/commit/0d22d6025ce2653ec9fb20a4e0745c79bda677da). We could probably simplify this to always using `--no-document` provided we bump the [minimum rubygems version to 2.0 in puppet's .gemspec](https://github.com/puppetlabs/puppet/blob/5.5.x/.gemspec#L20).

Reply:
Wouldn’t that introduce errors when managing gems in system rubies that ship with rubygems older than 2.0 and be a breaking change? I recognize that the potential is low ...

Reply:
For example:

```
[root@pe-agent ~]# cat /etc/redhat-release 
CentOS release 6.6 (Final)

[root@pe-agent ~]# yum install ruby rubygems -y -q

[root@pe-agent ~]# gem install colorize --no-document
ERROR:  While executing gem ... (OptionParser::InvalidOption)
    invalid option: --no-document
```

Reply:
Yeah, never mind about my comment. I forgot that the `gem` command we're executing is not necessarily the same version as the ruby version running puppet.